pkg/service/role.go

```diff
@@ -145,6 +145,12 @@ func (s *roleService) Create(ctx context.Context, role *rolev3.Role) (*rolev3.Ro
 		}
 	}
 
+	//validate basic mandatory permissions that should be part of all custom roles
+	if len(role.Spec.Rolepermissions) > 0 && (!utils.Contains(role.Spec.Rolepermissions, partnerR) ||
+		!utils.Contains(role.Spec.Rolepermissions, organizationR) || !utils.Contains(role.Spec.Rolepermissions, projectR)) {
```

Comment:
Just a nit for better [readability](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)
```suggestion
	if len(role.Spec.Rolepermissions) > 0 &&
		!(utils.Contains(role.Spec.Rolepermissions, partnerR) &&
			utils.Contains(role.Spec.Rolepermissions, organizationR) &&
			utils.Contains(role.Spec.Rolepermissions, projectR)) {
```